### PR TITLE
support zipkin endpoint configuration

### PR DESCRIPTION
**Pull request by Kuromesi**
This PR aims to support this feature request https://github.com/istio/istio/issues/53086.

Add `Endpoint` in zipkin extension provider to support customized endpoint.

### CODE REVIEW

File: mesh/v1alpha1/config.proto [hunk 1]
```diff
@@ -719,6 +719,10 @@ message MeshConfig {
       // Example: "zipkin.default.svc.cluster.local" or "bar/zipkin.example.com".
       string service = 1;
 
+      // Optional. Specifies the endpoint of Zipkin API.
+      // The default value is "/api/v2/spans".
+      string endpoint = 5;
+
```
Comment (zirain) [inline mesh/v1alpha1/config.proto:725]: move after `enable_64bit_trace_id`

Comment (Kuromesi) [reply mesh/v1alpha1/config.proto]: done

**Review by zirain — APPROVED** [review]

File: mesh/v1alpha1/config.proto [hunk 2]
```diff
@@ -729,6 +729,10 @@ message MeshConfig {
       // Optional. A 128 bit trace id will be used in Istio.
       // If true, will result in a 64 bit trace id being used.
       bool enable_64bit_trace_id = 4;
+
+      // Optional. Specifies the endpoint of Zipkin API.
+      // The default value is "/api/v2/spans".
+      string endpoint = 5;
```
[the remark this answers is not in the document]
Comment (Kuromesi) [reply mesh/v1alpha1/config.proto:735]: Istio zipkin config uses `CollectorEndpoint`, so I use `endpoint` to comply with that. Do you think `path` is better? 🤔

[the remark this answers is not in the document]
Comment (hzxuzhonghu) [reply mesh/v1alpha1/config.proto:735]: I remember some http extension provider use path, you can search in this repo

[the remark this answers is not in the document]
Comment (zirain) [reply mesh/v1alpha1/config.proto:735]: yeah, it's used in `MeshConfig_ExtensionProvider_HttpService`.

@Kuromesi let me switch to use `path`.

[the remark this answers is not in the document]
Comment (Kuromesi) [reply mesh/v1alpha1/config.proto]: Done, thanks for your comments!

**Review by howardjohn — APPROVED** [review]

**Pull request merged**